packages/block-editor/src/components/link-control/style.scss

```diff
@@ -3,6 +3,12 @@ $block-editor-link-control-number-of-actions: 1;
 .block-editor-link-control {
 	position: relative;
 	min-width: $modal-min-width;
+
+	.components-popover__content & {
```

Comment:
Why the LinkControl should be aware that. it can be used in a Popover?

Reply:
Are you saying the parent component should apply a custom class name?

Reply:
I don't know but it feels like:

 - The LinkControl shouldn't be aware of a Popover and should behave properly no matter where it's rendered (fluid maybe)
 - The Popover shouldn't be aware of its children of course but has smart logic to adapt to the viewport.

Now, I don't know the details of this issue to argue exactly what the fix should be.

Reply:
>The LinkControl shouldn't be aware of a Popover and should behave properly no matter where it's rendered (fluid maybe)

Agreed, this PR furthers that.

>The Popover shouldn't be aware of its children of course but has smart logic to adapt to the viewport.

This requires a bigger refactor of popovers.

Reply:
> Agreed, this PR furthers that.

It doesn't seem like it since there's an explicit mention of popover in the link control style.

> This requires a bigger refactor of popovers.

The popover should already have the logic to adapt the width to the viewport built-in. Maybe there's a bug there.

Reply:
>It doesn't seem like it since there's an explicit mention of popover in the link control style.

Which is used to remove pixel-based dimension constraints.

>The popover should already have the logic to adapt the width to the viewport built-in. Maybe there's a bug there.

Yeah, that logic is not very robust and it overflows the screen in most cases.

Reply:
> Which is used to remove pixel-based dimension constraints.

Yes, this is cool but can't we do it consistently without any mention of the popover class. I'm thinking the LinkControl can also be shown in narrow contexts independent of the Popover being present or not.

Reply:
No, because the way we can dampen the effect of the popover bug here does not necessarily work in other narrow contexts. We can only assume a `90vw` is OK because the popover renders on top of everything.

Reply:
Do you think a generic solution based on "react-resize-aware" would work here like the Placeholder component?

Reply:
Yes, or `react-spring/react-use-measure`.